k2/csrc/cuda_headers.cuh

```diff
@@ -0,0 +1,31 @@
+/**
+ * @brief
+ * Put cuda runtime headers here.
+ * With this file be included, code static analysis works.
+ *
+ * @copyright
+ * Copyright (c)  2020  Xiaomi Corporation (authors: Daniel Povey
+ *                                                   Haowen Qiu)
+ *
+ * @copyright
+ * See LICENSE for clarification regarding multiple authors
+ */
+
+#ifndef K2_CSRC_CUDA_HEADERS_H_
+#define K2_CSRC_CUDA_HEADERS_H_
+
+#ifdef __CUDA_ARCH__
+#define K2_CUDA_HOSTDEV __host__ __device__
+#else
+// then commented to compile
+#include <cuda.h>
```

Comment:
These header files are included automatically by nvcc.

Reply:
yeah, I guess so..as we can build without them.

Reply:
Right, as the header statement, it's for code static analysis. Could keep it for my convenience? And at any time, it's ready to be deleted.

Reply:
I'd rathe you take them out, it's confusing.  You could maybe remove them with a script prior to static analysis.